spec/orders.mediawiki

```diff
@@ -582,29 +584,65 @@ the checksum generated from their local copy of the epoch queue.
 ! field   !! size (bytes)  !! description
 |-
 | pimg    || string || hex-encoded preimage for the order's commitment
+|-
+| ackid   || int || the message ID to be used for the server's preimage acknowledgement
 |}
 
-==Match negotiation==
+The server will send an [[comm.mediawiki/#Acknowledgements|acknowledgement]] of
+their receipt of the preimage. The message ID of the acknowledgement will be set
+to the specified <code>ackid</code>. The signature message is the byte-encoded
+preimage.
 
-Swap negotiation details will be relayed through the DEX with a series of
-notifications.
-Both the DEX and the clients will need to serialize and sign the notification
-data. The originator includes their signature with the request, while the
-recipient will return an '''acknowledgement''', or a list of
-acknowledgements, as the <code>result</code> of their response payload.
+==Order Revocation==
+
+In a couple of situations, the server will revoke a client's orders.
+
+1. A client in violation of the rules of community conduct may have their existing orders revoked. In this case, the revocation will count against the client's cancellation statistics.
+
+2. If a client does not reconnect after a trade suspension with <code>persist = true</code>, the client's orders will be revoked. This revocation does not count against the client's cancellation statistics.
+
+The server will send a <code>revoke_order</code> request to the client when an
+order is revoked.
 
-'''Acknowledgement'''
+'''Request route:''' <code>revoke_order</code>, '''originator:''' DEX
 
+<code>payload</code>
 {|
-! field     !! type   !! description
+! field   !! type   !! description
+|-
+| orderid || string || order ID
 |-
-| matchid   ||  string  || the match ID
+| reason  || string || the reason the order is being revoked
 |-
-| sig       ||  string || hex-encoded signature of the notification data
+| sig     || string || DEX's hex-encoded signature of the serialized notification data. serialization described below
 |}
 
+'''Order revocation serialization'''
+
+{|
+! field   !! size (bytes)  !! description
+|-
+| orderid || 32 || the order ID
+|-
+| reason  || varies || the UTF-8 encoded reason
+|}
+
+The client will respond with an
+[[comm.mediawiki/#Acknowledgements|acknowledgement]].
```

Comment:
I'm also not sure if this ack serves a real purpose either.  What would the server do with the client's ack?  Much like with revoke_match, the server does nothing with the ack presently, just logging it.

Reply:
You okay with removing both acknowledgements and making these notifications instead?